src/c_api/c_api_common.h

```diff
@@ -35,16 +35,16 @@
 #include <string>
 
 /*! \brief  macro to guard beginning and end section of all functions */
-#define API_BEGIN() try {
+#define API_BEGIN() try { on_enter_api(__FUNCTION__);
```

Comment:
what's the overhead of this?

Reply:
Nothing if Profiler isn’t running and kAPI isn’t turned on